bin/cml-runner.js

```diff
@@ -379,11 +387,6 @@ const opts = yargs
     'cloud-ssh-private',
     'Custom private RSA SSH key. If not provided an automatically generated throwaway key will be used'
   )
-  .boolean('cloud-ssh-private-visible')
-  .describe(
-    'cloud-ssh-private-visible',
-    'Show the private SSH key in the output with the rest of the instance properties (not recommended)'
-  )
```

Comment:
what was the purpose of this? Was it meant to be hidden (`.hide()`)?

Reply:
> _What was the purpose of this?_

~Encourage bad security practices.~ Make debugging easier for those who don't want to generate and keep a local private key.

> _Was it meant to be hidden?_

Probably, it was added not so long ago and remained undocumented.

Reply:
@DavidGOrtega you ok with killing this? :skull:

Reply:
Which one was hide()?

I think that having boolean('cloud-ssh-private-visible') is useful. Why? not every project runs in public repos and they might want to have the SSH automatically generated and given in case they want to connect

Reply:
> _Which one was hide()?_

It was probably meant to be hidden, but we didn't hide it.

> _I think that having boolean('cloud-ssh-private-visible') is useful. Why? not every project runs in public repos and they might want to have the SSH automatically generated and given in case they want to connect_

In my opinion, providing this functionality to end users and — worse yet — encouraging its use could be a bad move at too many levels:

* No matter whether a repository is public or not, printing out plaintext secrets on any kind of log is not a good practice. Especially when those secrets can lead to the exfiltration of all the repository secrets; e.g. by logging in through SSH to the instance and reading the startup script or dumping the memory of the self-hosted runner process.

* We already provide the `--cloud-ssh-private=<pem>` option as a sane alternative, so we can generate and provide our own private keys without storing them anywhere in plaintext apart from our local computers and the cloud instances. Generating a local key and keeping it for future debug runs is even convenient.

* Users should not need to access the instance system directly from `cml-runner` as it's supposed to abstract the instance at the self-hosted runner level. If there is any need for interactive access, it should be preferably done — both for security and convenience — at the workflow level, where users can experiment with the exact environment where their workflows will run on.

I would wait to implement this functionality until we identify a real use case for it.